src/main/java/ch/cern/eam/wshub/core/services/contractmanagement/impl/EquipmentReservationAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package ch.cern.eam.wshub.core.services.contractmanagement.impl;
+
+import ch.cern.eam.wshub.core.client.InforContext;
+import ch.cern.eam.wshub.core.services.contractmanagement.EquipmentReservationAdjustmentService;
+import ch.cern.eam.wshub.core.services.contractmanagement.entities.EquipmentReservationAdjustment;
+import ch.cern.eam.wshub.core.tools.ApplicationData;
+import ch.cern.eam.wshub.core.tools.InforException;
+import ch.cern.eam.wshub.core.tools.Tools;
+import net.datastream.schemas.mp_fields.CUSTOMERRENTALADJUSTMENTID_Type;
+import net.datastream.schemas.mp_functions.mp7863_001.MP7863_GetCustomerRentalAdjustment_001;
+import net.datastream.schemas.mp_functions.mp7864_001.MP7864_AddCustomerRentalAdjustment_001;
+import net.datastream.schemas.mp_functions.mp7865_001.MP7865_SyncCustomerRentalAdjustment_001;
+import net.datastream.schemas.mp_functions.mp7866_001.MP7866_DeleteCustomerRentalAdjustment_001;
+import net.datastream.wsdls.inforws.InforWebServicesPT;
+
+import static ch.cern.eam.wshub.core.tools.DataTypeTools.isEmpty;
+
+public class EquipmentReservationAdjustmentServiceImpl implements EquipmentReservationAdjustmentService {
+    private final Tools tools;
+    private final InforWebServicesPT inforws;
+    private ApplicationData applicationData;
+
+    public EquipmentReservationAdjustmentServiceImpl(ApplicationData applicationData, Tools tools, InforWebServicesPT inforWebServicesToolkitClient) {
+        this.applicationData = applicationData;
+        this.tools = tools;
+        this.inforws = inforWebServicesToolkitClient;
+    }
+
+    @Override
+    public String createEquipmentReservationAdjustment(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        return tools.performInforOperation(context, inforws::addCustomerRentalAdjustmentOp, createAddOperation(context, equipmentReservationAdjustment))
+                .getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
```

Comment:
We need the defaults (code, Rtype, Rstatuscode). Please test that the Rstatuscode is not checked if Statuscode is supplied.